modules/sievefilters/site.js

```diff
@@ -222,6 +232,44 @@ $(function () {
             );
         });
 
+        $(document).on('click', '#edit_blocked_behavior', function(e) {
+            e.preventDefault();
+            let parent = $(this).closest('tr');
+            let elem = parent.find('#block_action');
+            let sender = $(this).closest('tr').children().html();
+            let scope = sender.startsWith('*@') ? 'domain': 'sender';
+
+            $('.dropdown').toggle();
+            Hm_Ajax.request(
+                [
+                    {'name': 'hm_ajax_hook', 'value': 'ajax_sieve_block_unblock'},
+                    {'name': 'imap_server_id', 'value': $(this).closest('.dropdown').prev().attr('mailbox_id')},
```

Comment:
Can we add the mailbox_id as part of the form, so we don't have to traverse the DOM to find out the ID? This traversal is fragile - as soon as we change the DOM for some other reason, this code will break.